packages/suite/src/views/onboarding/steps/SelectBackupType.tsx

```diff
@@ -0,0 +1,346 @@
+
+const SelectedOptionStyled = styled.div<{ $isDisabled: boolean }>`
+    cursor: ${({ $isDisabled }) => ($isDisabled ? undefined : 'pointer')};
+`;
+
+const SelectedOption = forwardRef<HTMLDivElement, SelectedOptionProps>(
+    ({ children, onClick, isDisabled }, ref) => (
+        <SelectedOptionStyled $isDisabled={isDisabled}>
+            <OptionStyled ref={ref} onClick={isDisabled ? undefined : onClick}>
+                <BackupIconWrapper>
+                    <Icon icon="BACKUP" size={24} />
+                </BackupIconWrapper>
+                {children}
+                <DownComponent />
+            </OptionStyled>
+        </SelectedOptionStyled>
+    ),
+);
+
+type OptionProps = {
+    children: ReactNode;
+    onSelect: () => void;
+    isChecked: boolean;
+};
+
+const Option = ({ children, onSelect, isChecked }: OptionProps) => (
+    <OptionStyled>
+        {children}
+        <Radio isChecked={isChecked} onClick={onSelect} />
+    </OptionStyled>
+);
+
+const FloatingSelectionsWrapper = styled.div<{ $elevation: Elevation }>`
+    z-index: ${zIndices.modal};
+    border-radius: ${borders.radii.sm};
+    box-shadow: ${({ theme }) => theme.boxShadowElevated};
+    background: ${mapElevationToBackground};
+`;
+
+type FloatingSelectionsProps = {
+    selected: BackupType;
+    onSelect: (value: BackupType) => void;
+    style: CSSProperties;
+};
+
+const FloatingSelections = forwardRef<HTMLDivElement, FloatingSelectionsProps>(
+    ({ selected, onSelect, style }, ref) => {
+        const { elevation } = useElevation();
+
+        return (
+            <FloatingSelectionsWrapper $elevation={elevation} ref={ref} style={style}>
+                <Option
+                    onSelect={() => onSelect('shamir-default')}
+                    isChecked={selected === 'shamir-default'}
+                >
+                    <OptionText>
+                        <Text variant={'tertiary'} typographyStyle="titleSmall">
+                            <Translation id={typesToLabelMap['shamir-default']} />
+                            <Badge variant="primary" inline margin={{ left: spacings.xxs }}>
+                                <Text typographyStyle="hint">
+                                    <Translation id="TR_ONBOARDING_BACKUP_TYPE_DEFAULT" />
+                                </Text>
+                            </Badge>
+                        </Text>
+                        <Text typographyStyle="hint">
+                            <Translation id="TR_ONBOARDING_SEED_TYPE_SINGLE_SEED_DESCRIPTION" />
+                        </Text>
+                    </OptionText>
+                </Option>
+                <Option
+                    onSelect={() => onSelect('shamir-advance')}
+                    isChecked={selected === 'shamir-advance'}
+                >
+                    <OptionText>
+                        <Text variant={'tertiary'} typographyStyle="titleSmall">
```

Comment:
nitpick: string is wrapped in curly braces, I would stick just with ''

Reply:
it would be great to add a prettier / eslint something that will autofix this. 

will you please look at it? :pray:

Reply:
True. I found [this](https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-brace-presence.md).

Let's not care about this for now. Just created a [proposal](https://www.notion.so/satoshilabs/Proposals-6296b830b8e94df889ba93f66f7964a5?pvs=4#7f57e9b0e5914726b048dfebaa478366) to add this rule and I will raise this pattern on code improvements session this week.